server/game/cards/01_SOR/units/EzraBridger.ts

```diff
@@ -0,0 +1,43 @@
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import { PlayType, TargetMode } from '../../../core/Constants';
+import AbilityHelper from '../../../AbilityHelper';
+
+
+export default class EzraBridger extends NonLeaderUnitCard {
+    protected override getImplementationId() {
+        return {
+            id: '9560139036',
+            internalName: 'ezra-bridger#resourceful-troublemaker'
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.addTriggeredAbility({
+            title: 'Look at the top card of your deck.',
+            when: {
+                onAttackCompleted: (event, context) => event.attack.attacker === context.source,
+            },
+            immediateEffect: AbilityHelper.immediateEffects.lookAt(
+                (context) => ({ target: context.source.controller.getTopCardOfDeck() })
+            ),
+            then: (thenContext) => {
+                const topCardOfDeck = thenContext.source.controller.getTopCardOfDeck();
+                return {
+                    title: 'Look at the top card of your deck.',
+                    thenCondition: (context) => context.source.controller.drawDeck.length > 0,
+                    targetResolver: {
+                        mode: TargetMode.Select,
+                        optional: true,
```

Comment:
I think we can consider removing optional: true here. The card does say "may" but in this case Leave it on top of the deck serves the same purpose.

Reply:
Yes Sir i'll remove the extra optional